### ISSUE TITLE
Elegant routing

### ISSUE THREAD

**Issue by ColemanGariety** [sarajs/Sara#32]
Right now, all modules needed in the client sandbox need to be required from within their respective actions, and global modules need to be referenced via the global object. This is awful and was introduced with the sandboxing, it should be refactored.

Right now:

```
var Data = require('./models/data')

app.get('/', function (req, res) {
  var MyView = require('./views/mine')
  MyView.render(Data.all())
  res.end(document.innerHTML)
})
```

The context and scoping is incredibly confusing.

**Comment by ColemanGariety** [comment]
Very similar to #27, could end up being the same issue.

**Comment by ColemanGariety** [comment]
Partially because I want Sara to fill the roll of Backbone, one should be allowed to drop any sort of framework they please into their Sara app.

The issue is that on the server, the global window object needs to be `Contextify`ed for each client, causing confusion.

A partial solution would be to break out the routes into separate files and write a special contextify require function so you know it has its own js context.

app.js

```
app.get('/', Sara.createContext('./routes/index'))
```

routes/index.js

```
var ThingView = require('../views/mine')
  , Thing = require('../models/data')

ThingView.render(Thing.all())
```

Routes would be synchronous, but could be run simultenously for each request, ideally. Not sure if that's possible. They wouldn't return anything, they'd just handle how the DOM should change on click.

**Comment by ColemanGariety** [comment]
`require()`s inside of the sandbox wouldn't use browserify, they'd grab the require from the global context if it existed, otherwise, it would somehow use node to require them within the sandbox context, this is the tricky part.

**Comment by ColemanGariety** [comment]
An alternative solution, and what's seeming more and more like the only one, is to make Sara run inside of herself.

new Sara.App() creates the app, then on request, it's run again within a Contextified window, which would eliminate the need for the above comment's hacky require solution, it could use the entire app scope for `require()`ing things. That would get messy fast, however.

**Comment by ColemanGariety** [comment]
Closed in 5ef287d02a
